add.go

```diff
@@ -112,6 +112,12 @@ func (b *Builder) Add(destination string, extract bool, options AddAndCopyOption
 	hostOwner := idtools.IDPair{UID: int(hostUID), GID: int(hostGID)}
 	dest := mountPoint
 	if destination != "" && filepath.IsAbs(destination) {
+		dir := filepath.Dir(destination)
+		if dir != "." && dir != "/" {
+			if err = idtools.MkdirAllAndChownNew(filepath.Join(dest, dir), 0755, hostOwner); err != nil {
+				return errors.Wrapf(err, "error creating directory %q", filepath.Join(dest, dir))
+			}
+		}
```

Comment:
if destination is a symlink does this work if the symlink target exists or again if the symlink target doesn't exist?

Reply:
Will check :+1:

Reply:
All current `bud.bats` tests pass, which includes `bud copy to symlink`, which tests copying to a [symlink that exists](https://github.com/containers/buildah/blob/master/tests/bud/dest-symlink/Dockerfile#L9). To be extra safe, I've verified that symlink is preserved:
```
STEP 7: RUN ls -ltr /etc
total 172
...
lrwxrwxrwx    1 root     root            16 Jun  8 15:24 hbase -> /usr/local/hbase
STEP 8: RUN ls -ltr /usr/local/hbase
total 4
-rw-r--r--    1 root     root           229 Jun  8 15:24 Dockerfile
```
This should cover the first case.

Reply:
For dangling symlink as destination, there wasn't a test present. The behaviour of Docker is that symlink is 'fixed' (i.e., path is resolved and created for `COPY`):
```
...
Step 3/5 : RUN ls -ltr /
 ---> Running in 42c15f182748
lrwxrwxrwx   1 root root    8 Jun  8 15:39 src -> /symlink
Removing intermediate container 42c15f182748
 ---> 310bbc52eaf7
Step 4/5 : COPY file /src/
 ---> c07aca50f4ef
Step 5/5 : RUN ls -ltr / && ls /src
 ---> Running in aaae074b5b4b
total 68
...
lrwxrwxrwx   1 root root    8 Jun  8 15:39 src -> /symlink
drwxr-xr-x   2 root root 4096 Jun  8 15:39 symlink
...
file
Removing intermediate container aaae074b5b4b
 ---> 74191ba0e053
Successfully built 74191ba0e053
```

This is the same behaviour as `buildah` exhibits with this PR applied:
```
...
STEP 3: RUN ls -ltr /
total 64
...
lrwxrwxrwx   1 root root    8 Jun  8 15:35 src -> /symlink
...
STEP 4: COPY file /src/file
STEP 5: RUN ls -ltr / && ls /src
total 68
...
lrwxrwxrwx   1 root root    8 Jun  8 15:35 src -> /symlink
drwxr-xr-x   2 root root 4096 Jun  8 15:35 symlink
...
file
STEP 6: COMMIT
Getting image source signatures
Copying blob 762d8e1a6054 skipped: already exists
Copying blob e45cfbc98a50 skipped: already exists
Copying blob d60e01b37e74 skipped: already exists
Copying blob b57c79f4a9f3 skipped: already exists
Copying blob 2512810ddbad done
Copying config b156c22ae4 done
Writing manifest to image destination
Storing signatures
--> b156c22ae4bbce876c9daa5d90ef7f40ba1ef125d45af10fa03f0062e58baffa
```
Will add a test for this.

Reply:
```
✓ bud copy to dangling symlink
```
Done! This should address this comment :+1: